src/libs/antares/study/variable-print-info.cpp

```diff
@@ -83,120 +82,110 @@ void variablePrintInfoCollector::add(const AnyString& name,
-    return str;
+    allVarsPrintInfo.at(varname).enablePrint(printStatus);
 }
 
-bool AllVariablesPrintInfo::setPrintStatus(std::string varname, bool printStatus)
+void AllVariablesPrintInfo::setMaxColumns(std::string varname, uint maxColumnsNumber)
 {
-    auto has_uppercase_name = [&](VariablePrintInfo* v) { 
-        std::string name = to_uppercase(v->name());
-        return to_uppercase(v->name()) == to_uppercase(varname); };
-
-    auto it = std::find_if(allVarsPrintInfo.begin(), allVarsPrintInfo.end(), has_uppercase_name);
-    if (it != allVarsPrintInfo.end())
-    {
-        (*it)->enablePrint(printStatus);
-        return true;
-    }
-    return false;
+    allVarsPrintInfo.at(varname).setMaxColumns(maxColumnsNumber);
 }
 
-void AllVariablesPrintInfo::setMaxColumns(std::string varname, uint maxColumnsNumber)
+std::string AllVariablesPrintInfo::name_of(unsigned int index) const
 {
-    auto has_name = [&](VariablePrintInfo* v) { return v->name() == varname; };
-    auto it = std::find_if(allVarsPrintInfo.begin(), allVarsPrintInfo.end(), has_name);
-    if (it != allVarsPrintInfo.end())
-        return (*it)->setMaxColumns(maxColumnsNumber);
+    return index_to_name.at(index);
 }
 
-void AllVariablesPrintInfo::prepareForSimulation(bool userSelection,
+void AllVariablesPrintInfo::prepareForSimulation(bool isThematicTrimmingEnabled,
                                                  const std::vector<std::string>& excluded_vars)
 {
-    assert(!isEmpty() && "The variable print info list must not be empty at this point");
-
     // Initializing output variables status
-    if (!userSelection)
+    if (not isThematicTrimmingEnabled)
         setAllPrintStatusesTo(true);
 
     for (const auto& varname : excluded_vars)
     {
-        const bool res = setPrintStatus(varname, false);
-        if (not res)
-            logs.info() << "Variable " << varname << " not found. Could not remove it";
+        setPrintStatus(varname, false);
     }
 
     // Counting zonal and link output selected variables
     countSelectedAreaVars();
     countSelectedLinkVars();
+    splitByPrintStatus();
 }
 
 bool AllVariablesPrintInfo::isPrinted(std::string var_name) const
 {
-    auto has_name = [&](VariablePrintInfo* v) { return v->name() == var_name; };
-    auto it = std::find_if(allVarsPrintInfo.begin(), allVarsPrintInfo.end(), has_name);
-    if (it != allVarsPrintInfo.end())
-        return (*it)->isPrinted();
-
-    // This point is not supposed to be reached (except in draft mode),
-    // because the searched variables should be found.
-    return true;
+    return allVarsPrintInfo.at(var_name).isPrinted();
 }
 
 void AllVariablesPrintInfo::setAllPrintStatusesTo(bool b)
 {
-    for (VariablePrintInfo* v : allVarsPrintInfo)
-        v->enablePrint(b);
+    for (auto pair : allVarsPrintInfo)
+        pair.second.enablePrint(b);
+}
+
+void AllVariablesPrintInfo::reverseAll()
+{
+    for (auto pair : allVarsPrintInfo)
+        pair.second.reverse();
```

Comment:
```suggestion
    for (auto& pair : allVarsPrintInfo)
        pair.second.reverse();
```
Otherwise you'd change a copy, not the elements themselves. See [this example](https://gist.github.com/flomnes/008920c58ba607c91d201b42a543507c).

Reply:
Thanks, this would have caused bug surely !